### DISCUSSION TITLE
Knowledge optimization

### DISCUSSION THREAD

**Discussion by sisuad** [langgenius/dify#3342] [Help] [answered]
When manually appending segments to the knowledge base, one must be aware that upon saving after the addition of keywords, those manually entered may be supplanted by those automatically generated by the model. Furthermore, when typing in Chinese input mode, pressing the 'Enter' key after a letter will result in the input method placing that letter on the screen, and the segmentation keyword recognition will mistakenly interpret the 'Enter' action as a command for an automatic line break.

version 0.6.1-fix1
---
知识库手动添加分段时，1.添加完关键词后保存，手动添加的关键词会被模型自动添加的关键词覆盖；2.在中文状态下输入某个字母，按回车键输入法会将该字母上屏，分段关键词会识别到回车自动换行 
版本0.6.1-fix1

**Comment by JohnJyong** [comment, marked as answer]
thanks for your feedback, wo have fixed the issue by below PR: https://github.com/langgenius/dify/pull/3351
